public/ledger.js

```diff
@@ -350,6 +361,85 @@ class Ledger {
       await transport.send(ledgerCLA, this.commands.SEND_TX, 2, 0);
     }
   }
+
+  /**
+   * Get signature for token on ledger
+   *
+   * @param {Object} data Data to send to Ledger
+   *   version + uid + len(symbol) + symbol + len(name) + name
+   *
+   * @return {Promise} Promise resolved when signature is received
+   */
+  signToken = async (data) => {
+    try {
+      const transport = await this.getTransport();
+      const result = await this.sendToLedgerOrQueue(transport, this.commands.SIGN_TOKEN, 0, 0, data);
+      return result.slice(0, -2);
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Send tokens and signatures to ledger before a tx
+   *
+   * @param {Object} tokens List of data to send to ledger
+   *   List of data with the format:
+   *     version + uid + len(symbol) + symbol + len(name) + name + len(signature) + signature
+   *
+   * @return {Promise} Promise resolved when signature is received
+   */
+  sendTokens = async (tokens) => {
+    const values = [];
+    try {
+      const transport = await this.getTransport();
+      for (let [index, data] of tokens.entries()) {
+        const value = await this.sendToLedgerOrQueue(transport, this.commands.SEND_TOKEN, index, 0, data);
+        // Compare response with ledger SW_OK
+        if (ledgerOK.compare(value) !== 0) {
+          // only return failures
+          values.push(data);
+        }
+      }
+      return values;
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Verify a token and signature with ledger
+   *
+   * @param {Object} data Data to send to Ledger
+   *   version + uid + len(symbol) + symbol + len(name) + name + len(signature) + signature
+   *
+   * @return {Promise} Promise resolved when signature is received
+   */
+  verifyTokenSignature = async (data) => {
+    try {
+      const transport = await this.getTransport();
+      await this.sendToLedgerOrQueue(transport, this.commands.VERIFY_TOKEN_SIGNATURE, 0, 0, data);
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Reset token signatures
+   *
+   * @param {Object} data Data to send to Ledger
+   *   version + uid + len(symbol) + symbol + len(name) + name + len(signature) + signature
+   *
+   * @return {Promise} Promise resolved when signature is received
+   */
+  resetTokenSignatures = async () => {
```

Comment:
We can reset only one signature, right? Should we change the name to singular? `resetTokenSignature`

Reply:
We can't reset a single signature, only all signatures

Reply:
I was a bit confused with the data parameter, it's good now.